gui/codeeditorstyle.cpp

```diff
@@ -0,0 +1,209 @@
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "codeeditorstyle.h"
+#include <QSettings>
+
+CodeEditorStyle::CodeEditorStyle(
+        const QColor& CtrlFGColor, const QColor& CtrlBGColor,
+        const QColor& HiLiBGColor,
+        const QColor& LnNumFGColor, const QColor& LnNumBGColor,
+        const QColor& KeyWdFGColor, const QFont::Weight& KeyWdWeight,
+        const QColor& ClsFGColor, const QFont::Weight& ClsWeight,
+        const QColor& QteFGColor, const QFont::Weight& QteWeight,
+        const QColor& CmtFGColor, const QFont::Weight& CmtWeight,
+        const QColor& SymbFGColor, const QColor& SymbBGColor,
+        const QFont::Weight& SymbWeight) :
+    widgetFGColor(CtrlFGColor),
+    widgetBGColor(CtrlBGColor),
+    highlightBGColor(HiLiBGColor),
+    lineNumFGColor(LnNumFGColor),
+    lineNumBGColor(LnNumBGColor),
+    keywordColor(KeyWdFGColor),
+    keywordWeight(KeyWdWeight),
+    classColor(ClsFGColor),
+    classWeight(ClsWeight),
+    quoteColor(QteFGColor),
+    quoteWeight(QteWeight),
+    commentColor(CmtFGColor),
+    commentWeight(CmtWeight),
+    symbolFGColor(SymbFGColor),
+    symbolBGColor(SymbBGColor),
+    symbolWeight(SymbWeight)
+{}
+
+bool CodeEditorStyle::operator==( const CodeEditorStyle& rhs ) const
+{
+    if( this->widgetFGColor != rhs.widgetFGColor ) return false;
+    if( this->widgetBGColor != rhs.widgetBGColor ) return false;
+    if( this->highlightBGColor != rhs.highlightBGColor ) return false;
+    if( this->lineNumFGColor != rhs.lineNumFGColor ) return false;
+    if( this->lineNumBGColor != rhs.lineNumBGColor ) return false;
+    if( this->keywordColor != rhs.keywordColor ) return false;
+    if( this->keywordWeight != rhs.keywordWeight ) return false;
+    if( this->classColor != rhs.classColor ) return false;
+    if( this->classWeight != rhs.classWeight ) return false;
+    if( this->quoteColor != rhs.quoteColor ) return false;
+    if( this->quoteWeight != rhs.quoteWeight ) return false;
+    if( this->commentColor != rhs.commentColor ) return false;
+    if( this->commentWeight != rhs.commentWeight ) return false;
+    if( this->symbolFGColor != rhs.symbolFGColor ) return false;
+    if( this->symbolBGColor != rhs.symbolBGColor ) return false;
+    if( this->symbolWeight != rhs.symbolWeight ) return false;
+    return true;
+}
+
+bool CodeEditorStyle::operator!=( const CodeEditorStyle& rhs ) const
+{
+    return !(*this == rhs);
+}
+
+CodeEditorStyle CodeEditorStyle::LoadSettings( QSettings *pSettings )
```

Comment:
Please use camelCase for function names.

I would prefer the name `settings` instead of `pSettings`.